### PR TITLE
Add legacy FTP/SFTP connector

### PR DESCRIPTION
**Pull request by jliao07**
Signed-off-by:

## Pre-Checklist

Note: Please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/bytedance/bitsail/blob/master/docs/contributing.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests.

## Purpose
Add legacy FTP/SFTP connector

## Approaches
Add code and integration tests

## Related Issues

_e.g._ Close #796

## New Behavior (screenshots if needed)

N/A

### CODE REVIEW

File: bitsail-common/src/main/java/com/bytedance/bitsail/common/option/ReaderOptions.java [hunk 1]
```diff
@@ -64,47 +66,138 @@ public interface ReaderOptions {
...
+        key(READER_PREFIX + "password")
             .noDefaultValue(String.class);
 
     ConfigOption<String> CONTENT_TYPE =
-        ConfigOptions.key(READER_PREFIX + "content_type")
+        key(READER_PREFIX + "content_type")
+            .noDefaultValue(String.class);
+  }
+
+  interface BaseFileReaderOptions extends BaseReaderOptions {
```
Comment (BlockLiu) [inline bitsail-common/src/main/java/com/bytedance/bitsail/common/option/ReaderOptions.java:113]: Options in BaseFileReaderOptions are never been used. Can we remove BaseFileReaderOptions?

File: bitsail-connectors/bitsail-connectors-legacy/bitsail-connector-ftp/src/test/resources/scripts/ftp_to_print.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        },
+        {
+          "name": "c2",
+          "type": "string"
+        }
+      ]
+    },
+    "writer": {
+      "class": "com.bytedance.bitsail.connector.legacy.print.sink.PrintSink",
+      "columns": [
```
Comment (BlockLiu) [inline bitsail-connectors/bitsail-connectors-legacy/bitsail-connector-ftp/src/test/resources/scripts/ftp_to_print.json:30]: Print writer does not need "columns". It's better to remove unused option.

File: docs/connectors/ftp/ftp.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+  - array
+
+## Parameters
+
+The following mentioned parameters should be added to `job.reader` block when using, for example: [ftp-connector-example](./ftp-example.md)
+
+### Necessary parameters
+
+| Param name   | Required | Optional value   | Description                                                                                   |
+| :----------- | :------- | :--------------- | :-------------------------------------------------------------------------------------------- |
```
Comment (BlockLiu) [inline docs/connectors/ftp/ftp.md:56]: Is the class name incomplete?

File: docs/connectors/ftp/ftp.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+| :--------------------- | :------- | :------ | ---- | :----------------------------------------------------------- |
+| connect_pattern           | No          | PASV if FTP, NULL if SFTP | PASV/PORT/     | In ftp mode, connect pattern can be PASV or PORT. In sftp mode, connect pattern is NULL |
+| time_out                  | No         | 5000ms               |                | Connection timeout                       |
+| enable_success_file_check | No         | True                 |                | Enabled by default, the job will not start if SUCCESS tag doesn't exist |
+| max_retry_time            | No         | 60                   |                | Max time to check for SUCCESS tag file |
+| retry_interval_ms         | No         | 60s                  |                | Retry interval to check for SUCCESS tag file |
+
+## <span id="jump_format">Supported formats</span>
+
+Support the following formats:
```
Comment (garyli1019) [inline docs/connectors/ftp/ftp.md:80]: Can we add the configuration was used to define the format here

**Review by garyli1019 — APPROVED** [review]
LGTM, left one minor comment.

File: bitsail-common/src/main/java/com/bytedance/bitsail/common/option/ReaderOptions.java (hunk 1 above)
Comment (jliao07) [reply bitsail-common/src/main/java/com/bytedance/bitsail/common/option/ReaderOptions.java:113]: It seems some of CSV/JSON options are needed by hadoop and ftp connector...

File: bitsail-connectors/bitsail-connectors-legacy/bitsail-connector-ftp/src/test/resources/scripts/ftp_to_print.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      "content_type": "csv",
+      "columns": [
+        {
+          "name": "c1",
+          "type": "string"
+        },
+        {
+          "name": "c2",
+          "type": "string"
+        }
```
Comment (ysamchu) [inline bitsail-connectors/bitsail-connectors-legacy/bitsail-connector-ftp/src/test/resources/scripts/ftp_to_print.json:25]: Can we add more types of column in addition to `string`?

File: bitsail-common/src/main/java/com/bytedance/bitsail/common/option/ReaderOptions.java (hunk 1 above)
Comment (jliao07) [reply bitsail-common/src/main/java/com/bytedance/bitsail/common/option/ReaderOptions.java:113]: Never mind, found those in RowParserOptions, will remove~

**Review by ysamchu — APPROVED** [review]
Overall LGTM. Leave minor comments on integration test coverage.

**Pull request merged**